src/metldata/builtin_transformations/infer_relation/model_transform.py

```diff
@@ -0,0 +1,154 @@
+) -> MultipleRelationSpec:
+    """Infer the multiplicity of an inferred relation based on the path.
+
+    Args:
+        path: The path to infer the multiplicity for.
+        schema: The underlying schema.
+
+    Returns:
+        The inferred multiplicity.
+    """
+    origin = target = False
+    # Traverse the path and check for multiplicity
+    for element in path.elements:
+        relation = get_relation(element, schema)
+        # If any multiplicity is observed, toggle the origin / source flag depending on
+        # the orientation of the relation.
+        if element.type_ == RelationPathElementType.ACTIVE:
+            if relation.multiple.origin:
+                origin = True
+            if relation.multiple.target:
+                target = True
+        else:
+            if relation.multiple.origin:
+                target = True
+            if relation.multiple.target:
+                origin = True
+        if origin and target:
+            break
+    return MultipleRelationSpec(origin=origin, target=target)
+
+
+def infer_mandatory_from_path(
+    path: RelationPath, schema: SchemaPack
+) -> MandatoryRelationSpec:
+    """Infer the mandatory property of an inferred relation based on the path.
+
+    Args:
+        path: The path to infer the mandatory property for.
+        schema: The underlying schema.
+
+    Returns:
+        The inferred mandatory property.
+    """
+    origin = target = True
+    # Traverse the path and check for mandatory
+    for element in path.elements:
+        relation = get_relation(element, schema)
+        # If either end is not mandatory, toggle the origin / source flag depending on
+        # the orientation of the relation.
+        if element.type_ == RelationPathElementType.ACTIVE:
+            if not relation.mandatory.origin:
+                origin = False
+            if not relation.mandatory.target:
+                target = False
+        else:
+            if not relation.mandatory.origin:
+                target = False
+            if not relation.mandatory.target:
+                origin = False
+        if not origin and not target:
+            break
+    return MandatoryRelationSpec(origin=origin, target=target)
+
+
+def infer_model_relation(
+    *, model: SchemaPack, transformation_config: InferRelationConfig
+) -> SchemaPack:
+    """Add inferred relations to a model.
+
+    Args:
+        model: The model based on SchemaPack to add the inferred relations to.
+        transformation_config: The config for inferring relations.
+
+    Returns:
+        The model with the inferred relation added.
+    """
+    updated_class_def: dict[str, ClassDefinition] = {}
+
+    path = transformation_config.relation_path
+
+    # This is the class that is being modified
+    source_class = transformation_config.relation_path.source
+
+    # This is the class that is referred by the path
+    target_class = transformation_config.relation_path.target
```

Comment:
Isn't source and target flipped here compared to how we used that in the old transformations?

Reply:
No, it is exactly the same with the previous infer relation implementation. On a path, target and source have different meanings than the old configuration nomenclature we used.